Add non-Hiver world overlay and flag Nh and Nk worlds as (minor)

mmarquee opened the pull request:
Adds a non-Hiver worlds overlay, and flags Tryylin and Prt'aow as Nh worlds.

inexorabletash commented on server/RenderContext.cs:
```diff
@@ -626,9 +626,37 @@ public void Render(XGraphics graphics)
-#endregion
+                #endregion
+
+                #region droyne
```
Should have a different `#region` name here.

inexorabletash commented on server/RenderContext.cs:
```diff
@@ -626,9 +626,37 @@ public void Render(XGraphics graphics)
+                    {
+                        if (world.HasCode("Nh"))
+                        {
+                            string glyph = "\u2605";
```
Maybe a different glyph? I'm partial to ✻ (U+273B TEARDROP-SPOKED ASTERISK) for Hiver things...

inexorabletash commented on server/RenderContext.cs:
```diff
@@ -626,9 +626,37 @@ public void Render(XGraphics graphics)
+                    solidBrush.Color = styles.nonHiverWorlds.textColor;
+                    foreach (World world in selector.Worlds)
+                    {
+                        if (world.HasCode("Nh"))
```
We may want to make this smarter, to avoid data changes; I don't think `Nh` is what we'd use in T5SS data. Rather, we could check for `HvFd` allegiance and add some generic sophont check to `World`? Even `world.HasCodePrefix("(")` might be sufficient.

mmarquee replied:
I guess that depends whether Nh means the same as Native Intelligent Life, or a significant non-Hiver population? How does T5SS cope with NIL from T5 world generation?

mmarquee replied:
Looks like T5 doesn't really say how to record NIL - other than (<name>), for example Regina has (Amindii)2. Looks like the above suggestion would work with some extra check of H1, H2, etc. which I guess should end up being HvH1, HvH2, etc. anyway.

mmarquee commented:
OK, I've changed all of my Nh (and Nk) codes to be (minor).

mmarquee commented:
The 3 remaining sectors - Phlask, Drakken and Lorspane don't have any Nh worlds, which is probably a product of when they were generated, rather than actual absence of NIL life in these sectors. They do have A and R zones, so maybe this could be extrapolated as being non-Hiver presence.

inexorabletash commented:
Looking good - I'll have a go with trying it out locally later today.

But I had the realization that this has turned into a "minor race homeworld" overlay. If we ditch the `HvFd` (etc) checks it would show the same data in that region of space but be generally useful elsewhere. What do you think of doing that?

inexorabletash commented:
Also, don't shy away from:

* Making fixes to Phlask, Drakken, Lorspane, or Extolian (since those don't have active developers)
* Convert any sectors to tab delimited or T5 column delimited, if that makes things easier than trying to adjust SEC columns

... but also no pressure to do either of those.

mmarquee commented:
Yes, a general NIL overlay - will this show the lack of NIL in Imperium and the flowering in Hiver and K'kree space ?

inexorabletash commented:
Okay, this looks good. I'd like to request two more changes and then I'll merge it.

* Change the URL param to "mh" for "minor homeworld".
* Eliminate the special case glyphs for hivers - doesn't seem worth it.

mmarquee commented:
OK, removed the Hiver specific bits of the overlay, changed the overlay name and convert the old formats of Phlask, Drakken and Lorspane to be second survey then flagged R and A travel zones inside the Federation as (minor) as well as some outside.

inexorabletash commented:
Okie dokie - it's merged and live! Thanks!

mmarquee commented:
Think something is up with Phlask, Drakken and Lorspane, did I commit the changes to sectors.xml that changes them to SecondSurvey?

inexorabletash commented:
You did, but I didn't push that file with the first publish. It's fixed,
they should appear now.

On Sun, Apr 3, 2016 at 10:40 AM, Mark Humphreys <notifications@github.com>
wrote:

> Think something is up with Phlask, Drakken and Lorspane, did I commit the
> changes to sectors.xml that changes them to SecondSurvey?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/inexorabletash/travellermap/pull/52#issuecomment-205019960>
>

mmarquee commented:
Actually it looks like the rename to .txt needs to be done to the project as well, these are still looking for .sec for these 3 sectors.

inexorabletash commented:
Ah yes, I only pop into Windows to work on the binary these days so don't pay much attention to the project file. Update coming shortly. 

> On Apr 3, 2016, at 10:55 AM, Mark Humphreys <notifications@github.com> wrote:
> 
> Actually it looks like the rename to .txt needs to be done to the project as well, these are still looking for .sec for these 3 sectors.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
>

inexorabletash commented:
While I'm here...

Whilst error checking, I noticed there are a bunch of length 5 or 6 routes
in Hiver space.

Or, if you're building the site, just pull up /admin/errors

/admin/codes is also useful when looking for glitches.

Want me to send you a list?

On Sun, Apr 3, 2016 at 11:11 AM, Joshua Bell <inexorabletash@gmail.com>
wrote:

> Ah yes, I only pop into Windows to work on the binary these days so don't
> pay much attention to the project file. Update coming shortly.
>
> On Apr 3, 2016, at 10:55 AM, Mark Humphreys <notifications@github.com>
> wrote:
>
> Actually it looks like the rename to .txt needs to be done to the project
> as well, these are still looking for .sec for these 3 sectors.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/inexorabletash/travellermap/pull/52#issuecomment-205021796>
>
>

mmarquee commented:
Great, I'll have a look at these, I'll raise an issue and then we can move on from this PR.
